backend/db_interface.py

```diff
@@ -0,0 +1,52 @@
+from pymongo import MongoClient, errors
+
+VERWIKI_DB_NAME = 'verwiki'
+TREES_TABLE_NAME = "radialTrees"
+
+def get_tree_by_id(id: int) -> object:
+    """
+    Returns the tree with the specified ID from the database, or 
+    an error, which is to be handled by the caller.
+    """
+    client = validate_and_retrieve_client() # Don't handle error here; allow it to bubble up
+    output = client[VERWIKI_DB_NAME][TREES_TABLE_NAME].find_one({"id": id})
+    client.close()
+    if output == None:
+        raise KeyError("No tree found with the id " + str(id))
```

Comment:
why raising KeyError here? 
Also, when trying to put dynamic values in strings, lets use f-strings (they make code very elegant, and get rid of excessively long lines by taking out + signs and str() calls)
The f-string would be 
`f"No tree found with the id {id}"`

Reply:
Here there is a key error since there was no tree found matching the given ID, i.e. the key was not in the DB. See here: https://realpython.com/python-keyerror/